File: chess++/Knight.cpp
#include "Knight.h"

Knight::Knight(const Player& player, const Position& position)
: ChessPiece(player, position) {
    this->name = "Knight";
    this->canJumpOverOthers = true;
    this->multipleMove = false;

    // Initializing possible moves for a knight
    this->possibleRegularMoves = {
        {1, 2}, {2, 1},
        {-1, 2}, {-2, 1},
        {-1, -2}, {-2, -1},
        {1, -2}, {2, -1}
    };
    // For Knight, capture moves are the same as regular moves
    this->possibleCaptureMoves = this->possibleRegularMoves;
}

File: chess/Knight.py
from .ChessPiece import ChessPiece

class Knight( ChessPiece ):
    def __init__(self, player, pos):
        super().__init__(player, pos)
        self.possibleRegularMoves = self.possibleCaptureMoves = [
            ( +1, +2 ), ( +2, +1 ),
            ( -1, +2 ), ( -2, +1 ),
            ( -1, -2 ), ( -2, -1 ),
            ( +1, -2 ), ( +2, -1 ),
        ]
        self.canJumpOverOthers = True
        self.multipleMove = False
        self.name = 'Knight'